src/RazorSdk/Tasks/StaticWebAssets/StaticWebAssetsGeneratePackManifest.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class StaticWebAssetsGeneratePackManifest : Task
+    {
+        // Since the manifest is only used at build time, it's ok for it to use the relaxed
+        // json escaping (which is also what MVC uses by default) and to produce indented output
+        // since that makes it easier to inspect the manifest when necessary.
+        private static readonly JsonSerializerOptions ManifestSerializationOptions = new()
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            WriteIndented = true
+        };
+
+        [Required]
+        public ITaskItem[] Assets { get; set; }
+
+        [Required]
+        public ITaskItem[] AdditionalPackageFiles { get; set; }
+
+        public ITaskItem[] AdditionalElementsToRemoveFromPacking { get; set; } = Array.Empty<ITaskItem>();
```

Comment:
This is used by static web assets V1 since they don't have a build manifest. When we compute the assets to pack in a cross-targeting environment, we add them here to make sure we remove them as content items from the actual packing so that they don't appear on `content` nor `contentFiles`